monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,459 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+
+import numpy as np
+
+from monai.config import KeysCollection
+from monai.transforms import SpatialCrop
+from monai.transforms.compose import MapTransform, Randomizable, Transform
+from monai.transforms.utils import generate_spatial_bounding_box
+from monai.utils import min_version, optional_import
+
+measure, _ = optional_import("skimage.measure", "0.14.2", min_version)
+distance_transform_cdt, _ = optional_import("scipy.ndimage.morphology", name="distance_transform_cdt")
+gaussian_filter, _ = optional_import("scipy.ndimage", name="gaussian_filter")
+
+
+# Transforms to support Training for Deepgrow models
+class FindAllValidSlicesd(Transform):
+    """
+    Find/List all valid slices in the label.
+    Label is assumed to be a 4D Volume with shape CDHW, where C=1.
+
+    Args:
+        label: key to the label source.
+        sids: key to store slices indices having valid label map.
+    """
+
+    def __init__(self, label: str = "label", sids: str = "sids"):
+        self.label = label
+        self.sids = sids
+
+    def _apply(self, label):
+        sids = []
+        for sid in range(label.shape[1]):  # Assume channel is first
+            if np.sum(label[0][sid]) == 0:
```

Comment:
Unless there's a reason for this I don't see, I'd recommend not using `continue` here and instead

```python
if np.sum(label[0][sid]) != 0:
    sids.append(sid)
```